File: sliding-window/121.Best-Time-to-Buy-and-Sell-Stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    curr := prices[0]
    res := 0

    for i := 1; i < len(prices); i++ {
        if prices[i] < curr {
            // lowest price is lower
            curr = prices[i]
        } else if (prices[i] - curr > res) {
            // profit is more
            res = prices[i] - curr
        }
    }

    return res
}

File: sliding-window/121.Best-Time-to-Buy-and-Sell-Stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        curr = prices[0]
        max_profit = 0

        for i in range(1, len(prices)):
            if (prices[i] < curr):
                curr = prices[i]
            elif (prices[i] - curr > max_profit):
                max_profit = prices[i] - curr

        return max_profit
